test/IRGen/Inputs/type_layout_dumper_other.swift

```diff
@@ -0,0 +1,64 @@
+public class SomeClass {}
+
+public protocol SomeProtocol {}
+
+@_fixed_layout
+public struct ConcreteFragileStruct {
+  var field: Int32
+
+  public struct NestedResilientStruct {}
+}
+
+@_fixed_layout
+public struct NonDependentFragileStruct<T : AnyObject> {
+  var field: T
+}
+
+@_fixed_layout
+public struct DependentFragileStruct<T> {
+  var field: T
+}
+
+public struct ConcreteResilientStruct {
+  var field: Int64
+}
+
+public struct NonDependentResilientStruct<T : AnyObject> {
+  var field: T
+}
+
+public struct DependentResilientStruct<T> {
+  var field: T
+
+  public struct NestedNonDependentResilientStruct {
+    var field: UnsafePointer<T>
+  }
+}
+
+public enum ConcreteResilientEnum {
+  case left(Int64)
+  case right(Int64)
+}
+
+public enum NonDependentResilientEnum<T : AnyObject> {
+  case some(UnsafePointer<T>)
+  case none
+}
+
+public enum DependentResilientEnum<T> {
+  case some(T)
+  case none
+}
+
+extension DependentResilientEnum {
+  public enum NestedNonDependentResilientEnum {
+    case some(UnsafePointer<T>)
+    case none
+  }
+}
+
+extension Int {
+  public struct NestedInExtension {
```

Comment:
What should I do about this case? Right now I'm printing the name of the type (qualified with the parent type if its nested), but that doesn't make sense when you have an extension of something from another module. Maybe the YAML should use mangled names instead?

Reply:
Mangled names sound correct to me. [That's what the reference dependencies do](https://github.com/apple/swift/blob/master/lib/FrontendTool/ReferenceDependencies.cpp#L172). I can see it being a little annoying to map back later, but it's probably the best choice.